Add CFS layer details (#38)
Cray-HPE/docs-uan: docs/portal/developer-portal/operations/Basic_UAN_Configuration.md

#      branch: "{{uan.working_branch}}"
```

> ### Required CFS Layers for UAN Configuration

Author: Addressed this comment.

### Slingshot Host Software (playbook: shs_{{default.network_type}}_install.yml)

> The first CFS Layer installs the Slingshot Host Software for the Slingshot network type of the system.  The `default.network_type` is is either `mellanox` or `cassini`.

Author: Addressed this comment.

### COS (playbook: cos-application.yml)

> The second CFS Layer runs the following roles from the `cos-config-management` VCS repository. Any configuration changes needed for these roles must be made in the `cos-config-management` `group_vars` or `host_vars` directories.

Reviewer: Reword the second sentence for clarity: 

```suggestion
The second CFS Layer runs the following roles from the `cos-config-management` VCS repository. Any configuration changes needed for these roles must be made in the `group_vars` or `host_vars` subdirectories of that repository.
```

Author: Addressed comments.

The following Ansible roles are run during UAN image configuration:

> - #### Standard UNIX configuration
>   - cos-config-map
>   - rsyslog
>   - localtime
>   - ntp
>   - limits
>   - kdump
> - #### Allow trust of CSM generated keys for elective passwordless SSH during image customization
> 
>   - trust-csm-ssh-keys
> - #### HPE Cray EX system configurations
> 
>   - overlay-preload
> - #### Install COS rpms
> 
>   - cos-services-install
> - #### DVS/LNET/FS
> 
>   - cray_lnet_install
>   - cray_dvs_install
>   - cray_lnet_load
>   - cray_dvs_load
> 
> The following Ansible roles are run during UAN post-boot configuration:
> 
> - #### Standard UNIX configuration
> 
>   - rsyslog
> - #### HPE Cray EX system configurations
> 
>   - ca-cert
>   - overlay-preload
> - #### GPU deploy support
> 
>   - cray_gpu_deploy

Reviewer: Let's change those level 4 headers to inline bold to avoid possible rendering issues. Add backticks for the role names too.

```suggestion
- **Standard UNIX configuration**
  - `cos-config-map`
  - `rsyslog`
  - `localtime`
  - `ntp`
  - `limits`
  - `kdump`
  
- **Allow trust of CSM generated keys for elective passwordless SSH during image customization**
  - `trust-csm-ssh-keys`
  
- **HPE Cray EX system configurations**
  - `overlay-preload`
  
- **Install COS rpms**
  - `cos-services-install`
  
- **DVS/LNET/FS**
  - `cray_lnet_install`
  - `cray_dvs_install`
  - `cray_lnet_load`
  - `cray_dvs_load`

The following Ansible roles are run during UAN post-boot configuration:

- **Standard UNIX configuration**
  - `rsyslog`
  
- **HPE Cray EX system configurations**
  - `ca-cert`
  - `overlay-preload`
  
- **GPU deploy support**
  - `cray_gpu_deploy`
```

Author: Addressed comments.

### CMS Layer (playbook: csm_packages.yml)

The third CFS Layer installs the the Cray Management System packages. These are normally not modified.

### Optional UAN Layer (playbook: set_nologin.yml)

> The Ansible roles involved in the UAN layer which runs the `set_nologin.yml` playbook touches the `/etc/nologin` file preventing non-root users from logging into the UAN while it is being configured.

Reviewer: 1. Since this layer is optional, can we give customers guidance as to when they should (or must) use this layer and when they can skip it?
2. Just to make sure customers don't miss the dependency, we should also state that if they use this optional layer, they **must** use the `unset_nologin.yml` as well or non-root users won't be able to log into the UANs.

Author: Addressed comments.

- [`uan_ldap`](uan_ldap.md): this optional role configures the connection to LDAP servers. To disable this role, the administrator must set 'uan_ldap_setup:no' in the 'uan-config-management' VCS repository.
- [`uan_hardening`](uan_hardening.md): This role configures site/customer-defined network security of UANs, for example, preventing ssh out of the UAN over the NMN to NCN nodes.

> The UAN roles in site.yml are required and must not be removed, with exception of `uan_ldap` if the site is using some other method of user authentication. The `uan_ldap` may also be skipped by setting the value of `uan_ldap_setup` to `no` in a `group_vars` or `host_vars` configuration file.  Configuration of this layer are made in the `uan-config-management` VCS repository.

Reviewer: ```suggestion
The UAN roles in `site.yml` are required and must not be removed, with exception of `uan_ldap` if the site is using some other method of user authentication. The `uan_ldap` may also be skipped by setting the value of `uan_ldap_setup` to `no` in a `group_vars` or `host_vars` configuration file.  Configuration of this layer is made in the `uan-config-management` VCS repository.
```

Author: Addressed comments.

### COS (playbook: cos-application-after.yml)

> This CFS Layer runs the following roles from the `cos-config-management` VCS repository. Any configuration changes needed for these roles must be made in the `cos-config-management` `group_vars` or `host_vars` directories.

Reviewer: Reword 2nd sentence for clarity.

```suggestion
This CFS Layer runs the following roles from the `cos-config-management` VCS repository. Any configuration changes needed for these roles must be made in the `group_vars` or `host_vars` subdirectories of that repository.
```

Author: Addressed comments.

The following Ansible roles are run during UAN image configuration:

> - rebuilt-initrd

Reviewer: ```suggestion
- `rebuilt-initrd`
```

Author: Addressed comments.

The following Ansible roles are run during UAN post-boot configuration:

> - configure_fs

Reviewer: ```suggestion
- `configure_fs`
```

Author: Addressed comments.
